src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -244,7 +245,25 @@ CHIP_ERROR pychip_DeviceController_DiscoverCommissioningLongDiscriminator(chip::
 
 void pychip_DeviceController_PrintDiscoveredDevices(chip::Controller::DeviceCommissioner * devCtrl)
 {
-    devCtrl->PrintDiscoveredDevices();
+    for (int i = 0; i < devCtrl->GetMaxCommissionableNodesSupported(); ++i)
+    {
+        const chip::Mdns::CommissionableNodeData * dnsSdInfo = devCtrl->GetDiscoveredDevice(i);
+        if (dnsSdInfo == nullptr)
+        {
+            continue;
+        }
+        ChipLogProgress(Discovery, "Device %d", i);
+        ChipLogProgress(Discovery, "\tHost name:\t\t%s", dnsSdInfo->hostName);
+        ChipLogProgress(Discovery, "\tLong discriminator:\t%u", dnsSdInfo->longDiscriminator);
+        ChipLogProgress(Discovery, "\tVendor ID:\t\t%u", dnsSdInfo->vendorId);
+        ChipLogProgress(Discovery, "\tProduct ID:\t\t%u", dnsSdInfo->productId);
+        for (int j = 0; j < dnsSdInfo->numIPs; ++j)
+        {
+            char buf[chip::Inet::kMaxIPAddressStringLength];
```

Comment:
Does kMaxIPAddressStringLength include nul-terminator? Suggest adding nul-terminator `+1` in `buf` to make it clear that no matter what, it's accounted-for.

Reply:
It does include space for the null terminator.  This is just `INET6_ADDRSTRLEN` from `<netinet/in.h>` (or equivalents for lwip).  Given this, I think there is already a convention here that a buffer so allocated does include space for null termination.  Admittedly, it is a bit obfuscated though with the define wrapped in our own named constant.